public/app/features/alerting/unified/components/rules/EditRuleGroupModal.tsx

```diff
@@ -1,18 +1,190 @@
 import { checkEvaluationIntervalGlobalLimit } from '../../utils/config';
 import { getRulesSourceName } from '../../utils/datasource';
 import { initialAsyncRequestState } from '../../utils/redux';
+import { parsePrometheusDuration } from '../../utils/time';
+import { DynamicTable, DynamicTableColumnProps, DynamicTableItemProps } from '../DynamicTable';
 import { EvaluationIntervalLimitExceeded } from '../InvalidIntervalWarning';
-import { evaluateEveryValidationOptions } from '../rule-editor/GrafanaEvaluationBehavior';
+import { evaluateEveryValidationOptions, MIN_TIME_RANGE_STEP_S } from '../rule-editor/GrafanaEvaluationBehavior';
+
+const MINUTE = '1m';
+interface AlertInfo {
+  alertName: string;
+  forDuration: string;
+  numberEvaluations: number;
+}
+function ForError({ message }: { message: string }) {
+  return <Badge color="orange" icon="exclamation-triangle" text={'Error'} tooltip={message} />;
+}
+
+const getNumberEvaluationsToStartAlerting = (forDuration: string, currentEvaluation: string) => {
+  const evalNumber = safeParseDurationstr(currentEvaluation);
+  const forNumber = safeParseDurationstr(forDuration);
+  if (evalNumber === 0) {
+    return 0;
+  } else {
+    const evaluationsBeforeCeil = forNumber / evalNumber;
+    return evaluationsBeforeCeil < 1 ? 0 : Math.ceil(forNumber / evalNumber) + 1;
+  }
+};
+
+export const isRulerGrafanaRuleDTO = (rule: RulerRuleDTO): rule is RulerGrafanaRuleDTO => 'grafana_alert' in rule;
+export const isAlertingRuleDTO = (rule: RulerRuleDTO): rule is RulerAlertingRuleDTO => 'alert' in rule;
+
+export const getAlertInfo = (alert: RulerRuleDTO, currentEvaluation: string): AlertInfo => {
+  const emptyAlert: AlertInfo = {
+    alertName: '',
+    forDuration: '0s',
+    numberEvaluations: 0,
+  };
+  if (isRulerGrafanaRuleDTO(alert)) {
+    return {
+      alertName: alert.grafana_alert.title,
+      forDuration: alert.for,
+      numberEvaluations: getNumberEvaluationsToStartAlerting(alert.for, currentEvaluation),
+    };
+  }
+  if (isAlertingRuleDTO(alert)) {
+    return {
+      alertName: alert.alert,
+      forDuration: alert.for ?? '1m',
+      numberEvaluations: getNumberEvaluationsToStartAlerting(alert.for ?? '1m', currentEvaluation),
+    };
+  }
+  return emptyAlert;
+};
+export const isValidEvaluation = (evaluation: string) => {
+  try {
+    const duration = parsePrometheusDuration(evaluation);
+
+    if (duration < MIN_TIME_RANGE_STEP_S * 1000) {
+      return false;
+    }
+
+    if (duration % (MIN_TIME_RANGE_STEP_S * 1000) !== 0) {
+      return false;
+    }
+
+    return true;
+  } catch (error) {
+    return false;
+  }
+};
+
+export const getIntervalForGroup = (
+  rulerRules: RulerRulesConfigDTO | null | undefined,
+  group: string,
+  folder: string
+) => {
+  const folderObj: Array<RulerRuleGroupDTO<RulerRuleDTO>> = rulerRules ? rulerRules[folder] : [];
+  const groupObj = folderObj?.find((rule) => rule.name === group);
```

Comment:
Could we improve naming here? `rule.name === group` looks a bit weird because we compare the group name to the rule name.

Reply:
trueee